ui/raidboss/popup-text.js

```diff
@@ -665,19 +665,42 @@ class PopupText {
         trigger.run(that.data, matches);
     };
 
-    // Run immediately?
-    if (!delay) {
-      f();
-      return;
+    let promise = null;
+
+    if ('promise' in trigger) {
+      if (typeof trigger.promise === 'function') {
+        promise = trigger.promise(data);
```

Comment:
Could you make this `trigger.promise(data, matches)`, so that `promise:` takes the same args as all the other trigger functions?

Reply:
Added `matches` to `test.js` and `README.md` with latest commit.